deeppavlov/dataset_iterators/file_paths_iterator.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Tuple, Iterator, Optional
+
+import numpy as np
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.data_learning_iterator import DataLearningIterator
+from deeppavlov.core.common.log import get_logger
+from deeppavlov.core.data.utils import chunk_generator
+
+log = get_logger(__name__)
+
+
+@register('file_paths_iterator')
+class FilePathsIterator(DataLearningIterator):
+    """Dataset iterator for datasetes like 1 Billion Word Benchmark.
+    It gets lists of file paths from the data dictionary and return lines from each file.
+
+    Args:
+        data: dict with keys ``'train'``, ``'valid'`` and ``'test'`` and values
+        seed: random seed for data shuffling
+        shuffle: whether to shuffle data during batching
+
+    Attributes:
+        shuffle: whether to shuffle data during batching
+        random: instance of ``Random`` initialized with a seed
+    """
+
+    def __init__(self, 
+                 data: dict, 
+                 seed: Optional[int] = None, 
+                 shuffle: bool = True,
+                 *args, **kwargs) -> None:
+        self.seed = seed
+        self.np_random = np.random.RandomState(seed)
+        super().__init__(data, seed, shuffle, *args, **kwargs)
+
+    def _shard_generator(self, shards, shuffle = False, random = None):
+        shards_to_choose = list(shards)
+        if shuffle:
+            random.shuffle(shards_to_choose)
```

Comment:
This is not a static method, you can use `self.np_random` and remove `random` from method arguments